aea/configurations/loader.py

```diff
@@ -202,6 +204,198 @@ def _validate(self, json_data: Dict) -> None:
+            if component_id in component_configurations:
+                raise ValueError(
+                    f"Configuration of component {component_id} occurs more than once."
+                )
+            component_configurations[component_id] = component_config
+
+        agent_configuration_obj.component_configurations = component_configurations
+        return agent_configuration_obj
+
+    def _dump_agent_config(
+        self, configuration: AgentConfig, file_pointer: TextIO
+    ) -> None:
+        """Dump agent configuration."""
+        agent_config_part = configuration.ordered_json
+        component_configurations = agent_config_part.pop("component_configurations")
+        self.validator.validate(instance=agent_config_part)
+        result = [agent_config_part] + [
+            OrderedDict(
+                name=id_.name,
+                author=id_.author,
+                version=id_.version,
+                type=id_.component_type.value,
+                **obj,
+            )
+            for id_, obj in component_configurations
+        ]
+        yaml_dump_all(result, file_pointer)
+
+    def _dump_component_config(self, configuration: T, file_pointer: TextIO) -> None:
+        """Dump component configuration."""
+        result = configuration.ordered_json
+        self.validator.validate(instance=result)
+        yaml_dump(result, file_pointer)
+
+    def _process_component_section(
+        self, i: int, component_configuration_json: Dict
+    ) -> Tuple[ComponentId, Dict]:
+        """
+        Process a component configuration in an agent configuration file.
+
+        It breaks down in:
+        - extract the component id
+        - validate the component configuration
+        - check that there are only configurable fields
+
+        :param i: the index of the component in the file.
+        :param component_configuration_json: the JSON object.
+        :return: the processed component configuration.
+        """
+        component_id, result = self._split_component_id_and_config(
+            i, component_configuration_json
+        )
+        self._validate_component_configuration(component_id, result)
+        self._check_only_configurable_fields(component_id, result)
+        return component_id, result
+
+    @staticmethod
+    def _split_component_id_and_config(
+        i: int, component_configuration_json: Dict
+    ) -> Tuple[ComponentId, Dict]:
+        """
+        Split component id and configuration.
+
+        :param i: the position of the component configuration in the agent config file..
+        :param component_configuration_json: the JSON object to process.
+        :return: the component id and the configuration object.
+        :raises ValueError: if the component id cannot be extracted.
+        """
+        result = deepcopy(component_configuration_json)
+        # author, name, version, type are mandatory fields
+        missing_fields = {"author", "name", "version", "type"}.difference(
```

Comment:
declare constant

Reply:
maybe a custom class for "AgentComponentConfiguration" is better, but at this stage, I don't think there's much value in adding that constant...